### PR TITLE
Support v2 "async" transactions returning multipart Arrow responses

### PR DESCRIPTION
**Pull request by NHDaly**
- Basic GET/PUT calls for the v2 /transactions API.
- Halfway decent return format for API functions.
- Basic mocked unit tests

### CODE REVIEW

File: src/api.jl [hunk 1]
```diff
@@ -19,11 +19,15 @@
 # Julia language.
 
 import JSON3
+import Arrow
+
+using Mocking: Mocking,@mock  # For unit testing, by mocking API server responses
```
Comment (NHDaly) [inline src/api.jl:24]:
```suggestion
using Mocking: Mocking, @mock  # For unit testing, by mocking API server responses
```

File: src/api.jl [hunk 2]
```diff
@@ -444,3 +541,21 @@ function load_model(ctx::Context, database::AbstractString, engine::AbstractStri
     actions = [_make_load_model_action(name, model) for (name, model) in models]
     return _post(ctx, PATH_TRANSACTION; query = query(tx), body = body(tx, actions...), kw...)
 end
+
+
+
+# --- utils -------------------------
+# TODO: Delete this once https://github.com/JuliaWeb/HTTP.jl/issues/816 is addressed:
```
Comment (vilterp) [inline src/api.jl:548]: explicitly note that this is copy/pasted from the lib

File: src/api.jl [hunk 3]
```diff
@@ -349,6 +353,99 @@ end
 # todo: when we have async transactions, add a variation that dispatches and
 #   waits .. consider creating two entry points for readonly and readwrite.
 
+function exec_v2(ctx::Context, database::AbstractString, engine::AbstractString, source; inputs = nothing, readonly = false, kw...)
```
Comment (vilterp) [inline src/api.jl:356]: change to `RAI.V2.exec`?

File: src/api.jl [hunk 4]
```diff
@@ -349,6 +353,99 @@ end
...
+end
+
+function _parse_multipart_results_response(msg)
+    # TODO: in-place conversion to Arrow without copying the bytes.
+    #   ... HTTP.parse_multipart_form() copies the bytes into IOBuffers.
+    parts = _parse_multipart_form(msg)
+    return _extract_multipart_results_response(parts)
+end
+function _extract_multipart_results_response(parts)
+    problems_idx = findfirst(p->p.name == "problems", parts)
```
Comment (vilterp) [inline src/api.jl:439]: change to hit `/problems`?

File: src/api.jl (hunk 3 above)
Comment (NHDaly) [reply src/api.jl:356]: 👍 but actually: i think this package is so early days, i'd rather just call this one _exec_, and rename the v1 exec to something else, maybe `RAI.V1.exec` or `RAI.exec_v1`!

(Or actually, naming this one `exec_async()` and then having one more wrapper called `exec()` that does the polling)

Comment (NHDaly) [reply src/api.jl:356]: For now i'm going to rename this to `exec_async()`, based on the above, i think.

File: src/api.jl (hunk 4 above)
Comment (NHDaly) [reply src/api.jl:439]: done. thanks!

File: src/api.jl (hunk 2 above)
Comment (NHDaly) [reply src/api.jl:548]: done, thanks

File: test/api.jl [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+            rsp = RAI.exec_v2(ctx, "engine", "database", "2+2")
+            @test rsp == JSON3.read("""{"id":"1fc9001b-1b88-8685-452e-c01bc6812429","state":"CREATED"}""")
+        end
+    end
+
+    @testset "sync response" begin
+        patch = make_patch(v2_fastpath_response)
+
+        apply(patch) do
+            data = make_arrow_table([4])
```
[the remark this answers is not in the document]
Comment (NHDaly) [reply test/api.jl:75]: done

**Review by vilterp — APPROVED** [review]

**Comment by NHDaly** [comment]
Thanks Pete! :)

**Pull request merged**